src/TransformOperationExecutor.ts

```diff
@@ -255,6 +257,26 @@ export class TransformOperationExecutor {
         }
     }
 
+    private attemptToUseTargetTypeConstructor(targetType: any) {
+        try {
+            return new targetType();
+        } catch (e) {
+            console.error(`${TRANSFORMATION_ERROR_PREFIX} - TransformOperationExecutor encountered error while attempting to call target type constructor with no parameters. The constructor of ${(targetType).name} may not be compatible with class-transformer by default, consider using a custom transformation function. Target constructor function was: `, targetType);
```

Comment:
Target type can be a `null` or `undefiend` here. Before trying to print we should check if it exists or not: 

```
const normalizedName = targetType && targetType.name ? targetType.name : targetType + ''
```

Also, no need to print the constructor because we already print the name of the class.

Reply:
I think actually seeing the constructor function could be valuable in many situations, especially for users working with classes that aren't defined within their application, it may give them context for why the constructor doesn't work, or how to write a custom transform function to handle it.  This was certainly the case for me when the firestore `Timestamp` constructor was giving me trouble.

It'd be nice if we could have a "verbose mode" config option somewhere that prints metadata and extra error logging info out to help people working with complex/difficult types.  Probably not in scope of this PR to add such a feature.

I'll push up update to fix the reference to targetType, but I'm going to leave the printing of the constructor in.  I'll remove it if you feel strongly that it shouldn't be there.

Reply:
> I think this use-case is non-existent. You will almost never end up using class-definitions outside of your app or shared repo.

It's certainly not common, but it's definitely not non-existent.
I've personally run into it with Firestore, they convert all `Date` values you store to `Timestamp` so I need to handle that class when reading data from my DB. I can see plenty of devs trying to use `moment` instead of `Date` (for all the same reasons Firestore uses `Timestamp`), and I could imagine it coming up when working with some specialty libraries for something like scientific computation, or an official SDK for something like Stripe might give you an instance of `Customer` rather than just a JSON object.

But, it's a pretty minor point, I'll just cut the additional logs so we can merge this.